src/backend/distributed/sql/udfs/citus_prepare_pg_upgrade/latest.sql

```diff
@@ -4,6 +4,21 @@ CREATE OR REPLACE FUNCTION pg_catalog.citus_prepare_pg_upgrade()
     SET search_path = pg_catalog
     AS $cppu$
 BEGIN
+    --
+    -- Drop existing backup tables
+    --
+    DROP TABLE IF EXISTS public.pg_dist_partition;
```

Comment:
I'm kind of wondering whether we should create a "citus_metadata_upgrade" schema that we can just drop in one go, though I still view this as an improvement.

Reply:
Yes, that would also work but it's a bigger change (though it would prevent this kind of thing: https://github.com/citusdata/citus/pull/4226). Let me know if want me to take that approach instead.